mmv1/third_party/terraform/services/storage/storage_managed_folder_test.go

```diff
@@ -0,0 +1,54 @@
+package storage_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccStorageManagedFolder_storageManagedFolderUpdate(t *testing.T) {
+	t.Parallel()
+	name := fmt.Sprintf("tf-test-managed-folder-%s", acctest.RandString(t, 10))
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccStorageManagedFolder_storageManagedFolderUpdate(name, false),
+			},
+			{
+				ResourceName:            "google_storage_managed_folder.folder",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"bucket", "force_destroy"},
+			},
+			{
+				Config: testAccStorageManagedFolder_storageManagedFolderUpdate(name, true),
+			},
```

Comment:
This test currently confirms that the virtual field can be updated but it doesn't confirm that the field affects the deletion of the folder.

Ideally this test could contain 2 additional steps that:
- adds a object into the managed folder, making it not empty
- removes the managed folder from the config (triggering a deletion) but the object is still in the config and therefore is present in the folder when it's deleted.

Would you be ok to add these steps?

---

For what it's worth, I can see that the new virtual field is present when the folder is deleted by the testing framework when it cleans up all the resources it creates during the test:

>DELETE /storage/v1/b/tf-test-managed-folder-uwyqek1f9n/managedFolders/managed%2Ffolder%2Fname%2F?allowNonEmpty=true

because of this, if you don't have time we could skip the new test steps being added

Reply:
Something like this would be used:

```tf
resource "google_storage_bucket_object" "object" {
  name   = "managed/folder/name/my-file.txt"
  content = "This file will affect the folder being deleted if allowNonEmpty=false"
  bucket = google_storage_bucket.bucket.name
}
```

Reply:
I've added the two extra steps. For the second step used [`testAccCheckGoogleStorageObject`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/storage/resource_storage_bucket_object_test.go#L509) which checks that the object is indeed in the bucket and not just in the state. Let me know if this is OK or if checking the state directly is better/preferable

I've also renamed the file to follow the same convention as the other tests in the storage folder.

Reply:
Checking that the file is still present in GCP after the delete operation might not be necessary as we're just wanting to test that the folder is being successfully deleted by the provider. I assume if the force_destroy value wasn't used in the URL for delete actions by the provider then the API would return an error saying that the folder isn't empty and cannot be deleted. Given that the test isn't failing due to this I think we're good!

I'm happy to either leave in or remove  the use of testAccCheckGoogleStorageObject.

Happy to approve this PR without this but here's an idea: Another improvement to the test could be to make the object be present from the start and have force_destroy=false on the folder. Then attempt to delete the folder in the next step, triggering an API error, but make the test step expect an error with `ExpectError` ([example](https://github.com/hashicorp/terraform-provider-google/blob/abcccfe983b864f0b7390c830879592822d54011/google/services/compute/resource_compute_region_backend_service_test.go#L85)). Then the test would update to force_destroy=true and re-attempt the delete.

Reply:
Done. Added the additional test step as suggested and removed testAccCheckGoogleStorageObject call as it makes the code a bit simpler.